junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/MutableExtensionRegistry.java

```diff
@@ -71,6 +71,8 @@
 	public static MutableExtensionRegistry createRegistryWithDefaultExtensions(JupiterConfiguration configuration) {
 		MutableExtensionRegistry extensionRegistry = new MutableExtensionRegistry(null);
 
+		TempDirectory.setConfiguration(configuration);
```

Comment:
Instead of doing this statically, I think we should rename `DEFAULT_EXTENSIONS` to `DEFAULT_STATELESS_EXTENSIONS`, remove `TempDirectory` from it, and call `extensionRegistry.registerDefaultExtension(new TempDirectory(configuration))`.

Reply:
Yep, that looks better.